ols/src/tools/tools.py

```diff
@@ -8,6 +8,18 @@
 logger = logging.getLogger(__name__)
 
 
+SENSITIVE_KEYWORDS = ["secret"]
+
+
+def raise_for_sensitive_tool_args(tool_args: dict) -> None:
+    """Check tool arguments for sensitive content and raise an exception if found."""
+    for key, value in tool_args.items():
+        if any(keyword in str(value).lower() for keyword in SENSITIVE_KEYWORDS):
+            raise ValueError(
+                f"Sensitive keyword in tool arguments {key}={value} are not allowed."
+            )
```

Comment:
@onmete - this seems to be very specific for OCP - what if other MCP servers , which may want the parameter secret ?

Reply:
Would you prefer to hardcode this check only if the mcp server is "openshift"?
I would cross this river once we get there. That might likely fall into "solved in l-stack" category.
Up to you.